zhusuan/distributions/special.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import division
+
+import tensorflow as tf
+from numpy import inf
+
+from zhusuan.distributions.base import Distribution
+from zhusuan.distributions.utils import assert_same_float_and_int_dtype, \
+    maybe_explicit_broadcast
+
+
+__all__ = [
+    'Empirical',
+    'Implicit',
+]
+
+
+class Empirical(Distribution):
+    """
+    The class of Empirical distribution.
+    See :class:`~zhusuan.distributions.base.Empirical` for details.
+
+    :param shape: A list or tuple describing the `batch_shape` of the distribution.
+        The entries of the list can either be int, Dimension or Tensor.
```

Comment:
Should the list allow Tensor?

Reply:
The issue, in fact, for using tensor is the one listed above. If the shape is a tensor `tf.TensorShape` does not work with it so I need to check for the exception.